examples/object_bounding_box.py

```diff
@@ -0,0 +1,35 @@
+"""
+Demonstrates visualizing object bounding boxes
+"""
+from pathlib import Path
+
+import trimesh
+
+import pygfx as gfx
+
+
+TEAPOT = Path(__file__).parent / "models" / "teapot.stl"
+teapot = trimesh.load(TEAPOT)
+
+scene = gfx.Scene()
+
+mesh = gfx.Mesh(
+    gfx.trimesh_geometry(teapot),
+    gfx.MeshPhongMaterial(),
+)
+mesh.rotation.set_from_euler(gfx.linalg.Euler(0.71, 0.91))
+scene.add(mesh)
+
+box_world = gfx.BoxHelper()
+box_world.set_object_world(mesh)
+scene.add(box_world)
+
+box_local = gfx.BoxHelper()
+box_local.set_object_local(mesh)
+mesh.add(box_local)  # note that the parent is `mesh` here, not `scene`
```

Comment:
These two modes confuse me a bit, does the box *have* to be a child of the mesh in the second case? What happens if its not? What happens if we do the same for the first case?

An idea, without having thought this through: instead of these methods to set a target, what if you simply made the box the child of any world object, and the box would size itself to the bounds of that object?

Reply:
The way I see it, a BoxHelper is just a box like an AxesHelper is just a set of axes. You can use it as you like; visualizing an object's bounding box is not its only purpose.

The methods are available just to make that one use case a little easier.

Reply:
That makes sense.

From my understanding `set_object_world` is more common/useful than `set_object_local`, right? The fact that in order for `set_object_local` to work correctly you must make it a child of the object seems a bit confusing to me. Is this use-case useful enough to justify a special method? If you think it is, we should probably document how to use it correctly.

Reply:
Why is that confusing? :) I think it's very elegant use of the scene graph functionality. It's precisely what it's intended for.

An example use case of local space bounding box visualization is collision detection with oriented (=local) bounding boxes.

Reply:
> I think it's very elegant use of the scene graph functionality.

Ok, I see what you mean. Though I still find it a bit confusing that there are two methods that look very similar, but with one you have to also make sure to put the the object in the right position. Anyway, good docstrings come a long way :)

Reply:
How about I add a boolean kwarg to choose local/world space and merge the methods?

Reply:
Yeah, maybe that helps because then there's a default. But if you prefer the current state I'm also fine with that. Maybe a good criterium is to see what version leads to simpler docs.